Article on What is rest api

D-stevpm opened the pull request:
#359 

- [ ] I have checked that my suggestion follows the Contribution guide

- [ ]  I have added which issue this PR closes

- [ ] I have 0 warnings from Danger.js

- [ ] If I have added images, they are in /files/ folder

prokopsimek commented on practices/what_is_rest_api:
```diff
@@ -0,0 +1,79 @@
+---
+slug: what_is_rest_api
```
```suggestion
slug: rest-api
```

prokopsimek commented on practices/what_is_rest_api:
```diff
@@ -0,0 +1,79 @@
+---
+slug: what_is_rest_api
+stages:
+  - research
```
```suggestion
  - development
```

prokopsimek commented on practices/what_is_rest_api:
```diff
@@ -0,0 +1,79 @@
+author_github_username: D-stevpm
+---
+
+# What is RESTful API
```
```suggestion
# REST API
```

prokopsimek commented on practices/what_is_rest_api:
```diff
@@ -0,0 +1,79 @@
+
+The majority of applications in use today are created for web and mobile platforms. Application Programming Interfaces (APIs) for the web have consequently gained importance. Among the different types of web APIs, REST (Representational State Transfer) API has become the most popular due to its simplicity, scalability, and flexibility.
+
+##What is an API 
```
```suggestion
### What is an API 
```

prokopsimek commented on practices/what_is_rest_api:
```diff
@@ -0,0 +1,79 @@
+
+![RESTful API](/files/what_is_rest_api.png)
+
+The majority of applications in use today are created for web and mobile platforms. Application Programming Interfaces (APIs) for the web have consequently gained importance. Among the different types of web APIs, REST (Representational State Transfer) API has become the most popular due to its simplicity, scalability, and flexibility.
```
```suggestion
**TL;DR**

The majority of applications in use today are created for web and mobile platforms. Application Programming Interfaces (APIs) for the web have consequently gained importance. Among the different types of web APIs, REST (Representational State Transfer) API has become the most popular due to its simplicity, scalability, and flexibility.
```

prokopsimek commented on practices/what_is_rest_api:
```diff
@@ -0,0 +1,79 @@
+*  An API has defined semantics or behavior to model the exchange of information effectively.
+*  Effective API design enables the extension to customers or third parties for business integration.
+
+##What is REST API?
```
```suggestion
## What is the REST API?
```

prokopsimek commented on practices/what_is_rest_api:
```diff
@@ -0,0 +1,79 @@
+
+Safe online information exchange between two computer systems is made possible by the RESTful API. To carry out various duties, the majority of business apps must interface with other internal and external applications. For instance, in order to automate invoicing and connect with an internal timesheet application, your internal accounts system must exchange information with your customers' banki ...
+
+###How to Implement RESTful API
```
Please add a section "Why You Might Want the REST API" and "Problems the REST API Helps to Solve" before this section. 

According to https://github.com/DXHeroes/knowledge-base-content/blob/master/CONTRIBUTING.md#content-composition

prokopsimek commented on practices/what_is_rest_api:
```diff
@@ -0,0 +1,79 @@
+*  **Easy to understand**: RESTful APIs are easy to understand due to their simplicity and uniform interface.
+*  **Easy to develop**: RESTful APIs are easy to develop due to their uniform interface and stateless nature, which reduces the complexity of the application.
+*  **Platform independence**: RESTful APIs can be used with any programming language and platform, making them highly interoperable.
+
```
Please, add missing sections "Common Pitfalls of the REST API" and "Resources for the REST API" according to https://github.com/DXHeroes/knowledge-base-content/blob/master/CONTRIBUTING.md#content-composition

D-stevpm replied:
Completed

prokopsimek commented on files/how_restful_api_works.png:
This image describes URLs in the form of `/allUsers`, `/newUser` and `/updateUser` but the best practice for REST API is to use `GET /users`, `POST /users` and `PUT /users`. Can you please change the image for example this (https://www.sitepoint.com/creating-restful-apis-express-4/#creating-the-api-for-the-movie-app) that describes the format of URL better? Thank you

prokopsimek reviewed (changes requested):
@D-stevpm Hi Steve, thank you for your article. The content is amazing! I am proposing to update some sections and please some sections according to the content composition are missing. Are you able to add those sections, please? Let me know if you need some help. :)

D-stevpm commented:
I will work on it and definitely let you know when I need assistance

D-stevpm commented:
Hi @prokopsimek I have updated the following section
- Why You Might Want the REST API
- Problems the REST API Helps to Solve
- Common Pitfalls of the REST API
- Resources for the REST API
- Also I changed the image as suggested, KIndly review it and let me know if there is something to adjust

prokopsimek commented:
@D-stevpm Hi, the article is amazing now! Is there any reason why you closed it? :) Thanks!

D-stevpm commented:
@prokopsimek Please let me know what you think about the new #396
